src/github.com/getlantern/pac-cmd/linux.c

```diff
@@ -31,7 +31,7 @@ int togglePac(bool turnOn, const char* pacUrl)
       // clear pac setting only if it's equal to pacUrl
       char* old_mode = g_settings_get_string(setting, "mode");
       char* old_pac_url = g_settings_get_string(setting, "autoconfig-url");
-      if (strcmp(old_mode, "auto") != 0 || old_pac_url == NULL || strcmp(old_pac_url, pacUrl) != 0 ) {
+      if (strcmp(old_mode, "auto") != 0 || strcmp(old_pac_url, pacUrl) != 0 ) {
```

Comment:
`g_settings_get_string` never returns NULL. Skip empty string check because if it's empty, `strcmp` will always return none 0.